cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt

```diff
@@ -46,6 +54,14 @@ open class CropImageActivity :
     private val takePicture = registerForActivityResult(ActivityResultContracts.TakePicture()) {
         if (it) onPickImageResult(latestTmpUri) else onPickImageResult(null)
     }
+    private val intentChooser =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { activityRes ->
+            if (activityRes.resultCode == Activity.RESULT_OK) {
+                (activityRes.data?.data ?: latestTmpUri)?.let { uri ->
```

Comment:
I think it actually never goes null because if it doesn't return a URI it will return the resultCode as Activity.RESULT_CANCELED. It's the same as the pickImageContract above but I'm not passing it back to the `onPickImageResult` if for any reason it tends to be null

Reply:
We cannot count with another place functionality for this piece of code. Why we don't call `onPickImageResult(uri)` and let it handle the null value?

Reply:
In case of intent chooser we can't determine whether the user has selected the camera app or a gallery app. If they select the camera app it uses `latestTmpUri` the hold the image while for a gallery app it returns the Uri in `activityRes.data?.data`. In case of the intent chooser it uses the same `registerForActivityResult` callback for both camera and gallery apps. That's why I check both `activityRes.data?.data` and `latestTmpUri` actually.

Reply:
We need to solve this ^^

Reply:
Can you suggest any solution? I'm a bit stuck on this one thing. Because the camera intent requires us to pass a uri (even on the existing code) while Gallery apps don't

Reply:
Actually ignore it @Canato I will work on it and try to perfect it.